stdlib/test/memory/test_memory.mojo

```diff
@@ -125,16 +125,16 @@ def test_memcmp():
 
 
 def test_memcmp_overflow():
-    var p1 = UnsafePointer[Int8].alloc(1)
-    var p2 = UnsafePointer[Int8].alloc(1)
+    p1 = UnsafePointer[Byte].alloc(1)
+    p2 = UnsafePointer[Byte].alloc(1)
     p1.store(-120)
     p2.store(120)
 
-    var c = memcmp(p1, p2, 1)
-    assert_equal(c, -1, "-120 is smaller than 120")
+    c = memcmp(p1, p2, 1)
+    assert_equal(c, 1)
 
     c = memcmp(p2, p1, 1)
-    assert_equal(c, 1, "120 is bigger than -120")
+    assert_equal(c, -1)
```

Comment:
@JoeLoser I think this small change actually revealed a bug. `memcpy` should treat the underlying buffer as `unsigned char`. [ref](https://cplusplus.com/reference/cstring/memcmp/).

Reply:
Yeah, I think we discovered this on a PR a few months ago that `memcpy` was always working with signed int8 which was recipe for error.

Reply:
Sorry I made a typo. `memcpy` using `Int8` is fine (at least from the ABI point of view). It's actually that `memcmp` was also using `Int8`, which is a logical error.